acados/ocp_nlp/ocp_nlp_sqp.c

```diff
@@ -1477,10 +1431,18 @@ int ocp_nlp_sqp(void *config_, void *dims_, void *nlp_in_, void *nlp_out_,
         qp_info *qp_info_;
         ocp_qp_out_get(mem->qp_out, "qp_info", &qp_info_);
         nlp_out->qp_iter = qp_info_->num_iter;
+        // printf("\nqp_iter = %d, sqp_iter = %d, max_sqp_iter = %d\n", nlp_out->qp_iter, sqp_iter, opts->max_iter);
         qp_iter = qp_info_->num_iter;
 
+        // save statistics of last qp solver call
```

Comment:
Better to always save the statistics for the qp here (i.e. remove the ones above about `qp_status` and `qp_iter`).

But you also need to add a safeguard similar to the one at (new) line 1448: this is because the memory for stat is fixed when the memory is created, and it is for `stat_m` lines. If the users afterwards in creases the max sqp iter, this still shouldn't result in overwriting memory around.